tests/ha/single_pod_restart/test_data_pod_restart.py

```diff
@@ -1872,129 +1846,166 @@ def test_continuous_writes_during_pod_restart(self):
-
-        LOGGER.info("Step 2: Check cluster status")
-        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_false(resp[0], resp)
-        LOGGER.info("Step 2: Cluster is in degraded state")
-
-        LOGGER.info("Step 3: Verify services that were running on pod %s are in offline state",
-                    pod_name)
-        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
-                                                           hostname=pod_host)
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 3: Verified services of %s are in offline state", pod_name)
-
-        remain_pod_list = list(filter(lambda x: x != pod_name, pod_list))
-        LOGGER.info("Step 4: Verify services status on remaining pods %s are in online state",
-                    remain_pod_list)
-        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=remain_pod_list, fail=False)
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 4: Verified services on remaining pods %s are in online state",
-                    remain_pod_list)
-
-        LOGGER.info("Step 5: Perform READ/WRITEs/VERIFY with variable object sizes in background")
+        LOGGER.info("STARTED: Test to verify READ/WRITE during data pod restart.")
+        LOGGER.info("STEP 1: Perform WRITEs/READs/Verify with variable object sizes")
+        LOGGER.info("Create 2 set of buckets to be used for writes and reads")
         users = self.mgnt_ops.create_account_users(nusers=1)
-        self.test_prefix = 'test-34076'
         self.s3_clean.update(users)
-        args = {'s3userinfo': list(users.values())[0], 'log_prefix': self.test_prefix,
-                'skipcleanup': True, 'nclients': 1, 'nsamples': 30}
-        thread = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,), kwargs=args)
-        thread.daemon = True  # Daemonize thread
-        thread.start()
-        LOGGER.info("Step 5: Performed READ/WRITEs/VERIFY with variable sizes objects in "
-                    "background.")
-
-        LOGGER.info("Step 6: Starting pod again by creating deployment using K8s command")
+        test_prefix_read = 'test-34076-read'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=test_prefix_read, skipcleanup=True,
+                                                    nsamples=2, nclients=5, skipread=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        test_prefix_write = 'test-34076-write'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=test_prefix_write, skipcleanup=True,
```

Comment:
add skipread

Reply:
no intentionally kept it that way to have write/read/verify on one set of buckets for which read/verify will be done again either on same bucket or diff bucket and first set just done write to do read/verify n background